Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionSidebar.h

```diff
@@ -0,0 +1,142 @@
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import <Foundation/Foundation.h>
+#import <WebKit/WKFoundation.h>
+
+@class WKWebExtensionContext;
+@class WKWebView;
+@protocol WKWebExtensionTab;
+@protocol WKWebExtension;
+
+WK_HEADER_AUDIT_BEGIN(nullability, sendability)
+
+/*!
+ @abstract This delegate notifies the browser that properties relevant to a currently-open sidebar have changed,
+ @discussion Implementing this delegate will allow the browser to react to changes to the sidebar object stemming from JavaScript API usage relating to the sidebar's web view or title.
+ */
+WK_CLASS_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA), visionos(WK_XROS_TBA))
+NS_SWIFT_NAME(WKWebExtension.ActiveSidebarDelegate)
+@protocol _WKWebExtensionActiveSidebarDelegate <NSObject>
+
+/*!
+ @abstract Called when the sidebar's web view has been updated.
+ @discussion This method is called when the sidebar's existing ``WKWebView`` is replaced by a different ``WKWebView``, in which case the browser
+ must query the sidebar in order to obtain a reference to this new ``WKWebView`` to display. This method is not necessarily called when the sidebar's panel
+ URL is updated: if the URL can be updated without replacing the ``WKWebView``, it will be.
+ */
+- (void)webViewWasUpdated;
+
+/*!
+ @abstract Called when the sidebar's title has been updated.
+ @discussion This method is called when the sidebar's title is updated through JavaScript API usage. When this method is called, the browser should query the current sidebar object for the new title in order to display it.
+ */
+- (void)titleWasUpdated;
+
+@end
+
+/*!
+ @abstract This delegate notifies the browser that properties relevant to a currently-closed sidebar have changed.
+ @discussion Implementing this delegate will allow the browser to react to changes to the sidebar object stemming from JavaScript API usage relating to the sidebar's enablement state or icon.
+ */
+WK_CLASS_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA), visionos(WK_XROS_TBA))
+NS_SWIFT_NAME(WKWebExtension.SidebarEnablementDelegate)
+@protocol _WKWebExtensionInactiveSidebarDelegate <NSObject>
+
+/*!
+ @abstract Called when the sidebar has changed from a disabled state to an enabled state.
+ */
+- (void)sidebarWasEnabled;
+
+/*!
+ @abstract Called when the sidebar has changed from an enabled state to a disabled state.
+ */
+- (void)sidebarWasDisabled;
+
+/*!
+ @abstract Called when the sidebar's icon has been updated
+ */
+- (void)iconWasUpdated;
+
+@end
```

Comment:
I wasn't expecting delegates per sidebar. Hmm. This is likely too granular.

I was imagining one or two methods on the existing extension controller delegate.

Something like:

```objc
/*!
 @abstract Called when the sidebar for a specific tab is updated.
 @param controller The web extension controller initiating the update.
 @param sidebar The sidebar object that has been updated.
 @param context The context within which the web extension is running.
 @discussion This method is called whenever the sidebar content is updated due to script actions or other triggers within the extension. 
 This can include changes to the sidebar's appearance, content, or state. Implementing this method is necessary if the app intends to 
 react to dynamic updates in the sidebar and ensure the UI is synchronized with the latest sidebar state.
 */
- (void)_webExtensionController:(WKWebExtensionController *)controller sidebarDidUpdate:(_WKWebExtensionSidebar *)sidebar forExtensionContext:(WKWebExtensionContext *)context;
```